mapusaurus/hmda/views.py

```diff
@@ -42,14 +41,21 @@ def loan_originations(request):
     else: 
         return HttpResponseBadRequest("Missing one of lender, action_taken, lat/lon bounds or geoid.")
     query = query.values('geo__geoid', 'geo__census2010households__total').annotate(volume=Count('geo__geoid'))
-    return query; 
+    return query 
     
-def get_peer_list(lender, metro):
+""" Returns a list of peers for a lender+metro combination based on fha_bucket and lar count. 
+    Allows to exclude selected institution/lender and order by institution's assets
+""" 
+def get_peer_list(lender, metro, exclude, order_by):
     loan_stats = lender.lendingstats_set.filter(geo_id=metro.geoid).first()
     if loan_stats:
-        percent_50 = loan_stats.lar_count * .50
+        percent_50 = round(loan_stats.lar_count * .50)
         percent_200 = loan_stats.lar_count * 2.0
-        peer_list = LendingStats.objects.filter(geo_id=metro.geoid, fha_bucket=loan_stats.fha_bucket, lar_count__range=(percent_50, percent_200)).exclude(institution=lender)
+        peer_list = LendingStats.objects.filter(geo_id=metro.geoid, fha_bucket=loan_stats.fha_bucket, lar_count__range=(percent_50, percent_200)).select_related('institution')
+        if exclude:
+            peer_list = peer_list.exclude(institution=lender)
+        if order_by:
+            peer_list = peer_list.order_by('-institution__assets')
```

Comment:
It seems like this might be a better fit as a function inside the lender (Institution?) class?

A quick summary of the intended use of view functions - 'A view function, or view for short, is simply a Python function that takes a Web request and returns a Web response.'